core/rfb.js

```diff
@@ -34,18 +34,18 @@ const DISCONNECT_TIMEOUT = 3;
 const DEFAULT_BACKGROUND = 'rgb(40, 40, 40)';
 
 export default class RFB extends EventTargetMixin {
-    constructor(target, url, options) {
+    constructor(target, urlOrChannel, options) {
         if (!target) {
             throw new Error("Must specify target");
         }
-        if (!url) {
-            throw new Error("Must specify URL");
+        if (!urlOrChannel) {
+            throw new Error("Must specify URL or WebSocket / RTCDataChannel");
         }
 
         super();
 
         this._target = target;
-        this._url = url;
+        this._urlOrChannel = urlOrChannel;
```

Comment:
Can we avoid spreading the ambiguity beyond the constructor? I would prefer keeping `url` and setting it to  `null` when we don't know it.